samples/explore_favorites.py

```diff
@@ -0,0 +1,158 @@
+# This script demonstrates how to get all favorites, or add/delete a favorite. 
+# favorites_item and favorites_endpoint may be updated
+# these method examples were made prior to any updates to the favorites code
+
+import argparse
+import logging
+import tableauserverclient as TSC
+
+def main():
+
+    parser = argparse.ArgumentParser(description="Explore datasource functions supported by the Server API.")
+    # Common options; please keep those in sync across all samples
+    parser.add_argument("--server", "-s", required = True, help = "server address")
+    parser.add_argument("--site", "-S", help = "site name")
+    parser.add_argument(
+        "--token-name", "-p", required = True, help = "name of the personal access token used to sign into the server"
+    )
+    parser.add_argument(
+        "--token-value", "-v", required = True, help = "value of the personal access token used to sign into the server"
+    )
+    parser.add_argument(
+        "--logging-level",
+        "-l",
+        choices = ["debug", "info", "error"],
+        default = "error",
+        help = "desired logging level (set to error by default)",
+    )
+    
+    args = parser.parse_args()
+
+    # Set logging level based on user input, or error by default
+    logging_level = getattr(logging, args.logging_level.upper())
+    logging.basicConfig(level = logging_level)
+
+    # SIGN IN
+    tableau_auth = TSC.PersonalAccessTokenAuth(args.token_name, args.token_value, site_id = args.site)
+    server = TSC.Server(args.server, use_server_version = True)
+    with server.auth.sign_in(tableau_auth):
+
+        def get(self, user_item: "UserItem", req_options: Optional["RequestOptions"] = None) -> None:
+            # specify user
+            user = server.users.get_by_id('96d4fd52-5169-4de5-hapa-fe90fae2018p')
+            # get all favorites on site
+            print("Getting all favorites for user: {}".format(user.id))
+            server.favorites.get(user)
+            # return favorites
+            print("Favorites for user: {}".format(user.id))
+            print(user.favorites)
+
+        def add_favorite_workbook(self, user_item: "UserItem", workbook_item: "WorkbookItem") -> None:
+            # get list of workbooks
+            all_workbook_items, pagination_item = server.workbooks.get()
+            # specify user
+            user = server.users.get_by_id('96d4fd52-5169-4de5-hapa-fe90fae2018p')
```

Comment:
Should extract the user id and assign it once, not hardcode it in multiple places